### ISSUE TITLE
IntaRNAsTar Exception raised

### ISSUE THREAD

**Issue by meichelman** [BackofenLab/IntaRNA#219]
Hi,

I am running IntaRNAsTar with a genome and list of sRNAs both in fasta format. I ran the following on the command line:

IntaRNAsTar -t target.file -q query.file --out output.file -m M -n 3 --threads 4 --outCsvCols id1,id2,subseq1,subseq2,start1,end1,start2,end2,bpList,P_E

The program did not successfully complete, and instead output the following warning:

"# WARNING : Exception raised for : #thread 1 #target 0 #query 7 : IndexRangeList::push_back(14350-40535) violates order given last range = 1985706-2032352"
"# WARNING : Exception raised : std::exception

  ==> Please report (including input) to the IntaRNA development team! Thanks!
  "
I have previously run a similar command on the same data and it worked successfully. The only differences are the outCsvCols, specifying the mode, specifying the number of subinteractions, and specifying the number of threads.

**Comment by martin-raden** [comment]
Hi,
thanks for the bug report. The error is most likely a processing order issue due to multithreading where some predictions take too long such that the order is not met..

Could you provide the input? Since without it is near to impossible to check what's going on.

Best,
Martin

**Comment by meichelman** [comment]
Yes, I am attaching the 2 files. The target file is "TIGR4.fasta" and the query file is "putative_sRNAs.fasta". It will not let me attach the files as .fasta so I am attaching them as .txt.

The command works just fine if I do not specify the number of threads. I have not tried a number other than the default and 4.

Thanks for the help.
[putative_sRNAs.txt](https://github.com/user-attachments/files/15594507/putative_sRNAs.txt)
[TIGR4.txt](https://github.com/user-attachments/files/15594516/TIGR4.txt)
